tests/basictest.cpp

```diff
@@ -55,3 +52,74 @@ TEST_F(BasicTest, NoMainModule) {
 	driver drv;
 	ASSERT_THROW(drv.parse_string(in), NoMainModuleException);
 }
+
+TEST_F(BasicTest, TwoModules) {
+	std::string in = "module useless {\n"
+									 "private: [x, y];\n"
+									 "output: z;\n"
+									 "concentrations: {\n"
+									 "x := 50;\n"
+									 "y := 30;\n"
+									 "}\n"
+									 "reactions: {\n"
+									 "x + y -> x + y + z;\n"
+									 "z -> 0;\n"
+									 "}\n"
+									 "}\n"
+									 "module main {\n"
+									 "private: [x, y];\n"
+									 "output: z;\n"
+									 "concentrations: {\n"
+									 "x := 50;\n"
+									 "y := 30;\n"
+									 "}\n"
+									 "reactions: {\n"
+									 "x + y -> x + y + z;\n"
+									 "z -> 0;\n"
+									 "}\n"
+									 "}\n";
+	std::string out = "#!/usr/bin/env crnsimul\n"
+										"main_x := 50;\n"
+										"main_y := 30;\n"
+										"main_x + main_y -> main_x + main_y + main_z;\n"
+										"main_z -> 0;\n";
+	driver drv;
+	ASSERT_EQ(drv.parse_string(in), 0);
+	EXPECT_EQ(drv.out, out);
```

Comment:
do we have to put the "#!/usr/bin/env crnsimul\n" ourselves?

Reply:
No, the compiler does that automatically. It's in the driver I believe